钙钛矿太阳能电池人工神经网络ppsci实现

ZcwDev opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/PaddleScience/pull/96 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
New features
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
1. Add perovskite solar cells neural network model (`psc_nn.py`)
2. Add configuration file for the model (`psc_nn.yaml`)
3. Implement hyperparameter optimization using Optuna
4. Add evaluation and visualization functionality
### Describe
<!-- Describe what this PR does -->
This PR implements a neural network model for predicting the short-circuit current density (Jsc) of perovskite solar cells. Key features include:

1. Model Architecture:
   - Multi-layer perceptron (MLP) with configurable layers and units
   - Hyperparameter optimization using Optuna
   - Support for multiple optimizers (Adam, RMSProp, SGD)

2. Training Features:
   - Weighted MSE loss function
   - Learning rate scheduling with warmup
   - Evaluation during training
   - Model checkpointing

3. Evaluation:
   - Multiple metrics (RMSE, R2, MAPE)
   - Results visualization
   - Model validation

4. Configuration:
   - YAML-based configuration
   - Flexible hyperparameter settings
   - Separate train/eval modes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/PaddleScience?pullRequest=1055) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/PaddleScience?pullRequest=1055) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: ZcwDev<br/>:x: 张成威<br/><hr/>**张成威** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/PaddleScience?pullRequest=1055) it.</sub>

HydrogenSulfate reviewed (commented):
感谢提交PR，有几点PR相关的问题：
1. 请使用pre-commit对代码进行格式化
![image](https://github.com/user-attachments/assets/1762d0f0-535c-488f-9f6d-29518755744a)
2. 除了mkdocs之外，请同步更新README.md和index.md内的案例列表

HydrogenSulfate commented on docs/zh/examples/params/solar_cell_pretrained.pdparams:
该文件已经上传，请将所有相关的路径替换为以下url：
<https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams>

HydrogenSulfate commented on examples/ML_Pipeline/conf/nn_optuna_ppsci.yaml:
文件名建议改为相关明确的，nn_optuna_ppsci看不出来是什么案例的配置文件

HydrogenSulfate commented on examples/ML_Pipeline/NN.py:
```diff
@@ -0,0 +1,350 @@
+   )
+
+def create_validator(input_dict, batch_size, name="validator"):
+   """创建评估器"""
```
注释建议改为英文

HydrogenSulfate commented on docs/zh/examples/NN.md:
文件名建议改为案例相关的名字，比如`perovskite_solar_cells`之类的

HydrogenSulfate commented on docs/zh/examples/images/ML/En.png:
文件名使用小写，并且改成有意义的名字，如overview、pipeline

HydrogenSulfate commented on examples/ML_Pipeline/NN.py:
同样的文件名问题

HydrogenSulfate reviewed (commented):
ML_Pipeline文件夹名字建议改一下，如perovskite_solar_cells

HydrogenSulfate commented on examples/ML_Pipeline/NN.py:
```diff
@@ -0,0 +1,350 @@
+               "label": input_dict["label"],
+           },
+           "batch_size": batch_size,
+           "sampler": {
+               "name": "BatchSampler",
+               "drop_last": False,
+               "shuffle": False,
+           },
```
69-73这个sampler字段在drop_last和shuffle都是false情况下，可以删掉

HydrogenSulfate commented on examples/ML_Pipeline/conf/nn_optuna_ppsci.yaml:
```diff
@@ -0,0 +1,34 @@
+mode: "train"
+seed: 42
+output_dir: "./output"
+
```
文件开头建议加上如下的配置：https://github.com/PaddlePaddle/PaddleScience/blob/ee2d77cd99fc57f8ecf39d1a072a2b10901575d0/examples/allen_cahn/conf/allen_cahn_piratenet.yaml#L1-L24

HydrogenSulfate commented on examples/ML_Pipeline/conf/nn_optuna_ppsci.yaml:
```diff
@@ -0,0 +1,34 @@
+eval:
+  batch_size: 64
+  eval_with_no_grad: true
+  pretrained_model_path: null  # 预训练模型路径，可以是本地路径或URL
```
注释建议改为英文

HydrogenSulfate commented on examples/ML_Pipeline/NN.py:
```diff
@@ -0,0 +1,350 @@
+   return model
+
+def train(cfg: DictConfig):
+    print("Starting training...")
+    ppsci.utils.misc.set_random_seed(cfg.seed)
+    logger.init_logger("ppsci", osp.join(cfg.output_dir, f"{cfg.mode}.log"), "info")
+    
+    if not os.path.exists(cfg.output_dir):
+        os.makedirs(cfg.output_dir)
+
```
这里可以删除，配置文件中添加了callback字段后，会自动创建output_dir

HydrogenSulfate commented on examples/ML_Pipeline/NN.py:
```diff
@@ -0,0 +1,350 @@
+            output_dir=cfg.output_dir,
+            epochs=cfg.train.epochs,
+            iters_per_epoch=len(X_train) // cfg.train.batch_size,
+            eval_during_train=True,
+            eval_freq=5,
+            save_freq=10,
+            eval_with_no_grad=True,
+            log_freq=50,
```
这些字段建议移动到yaml中，由yaml控制

HydrogenSulfate commented on examples/ML_Pipeline/NN.py:
```diff
@@ -0,0 +1,350 @@
+        epochs=cfg.train.epochs,
+        iters_per_epoch=len(X_train) // cfg.train.batch_size,
+        eval_during_train=cfg.train.eval_during_train,
+        eval_freq=5,
+        save_freq=10,
+        eval_with_no_grad=cfg.eval.eval_with_no_grad,
+        log_freq=50,
```
同上

HydrogenSulfate commented on examples/ML_Pipeline/NN.py:
```diff
@@ -0,0 +1,350 @@
+    visualize_results(solver, X_val, y_val, cfg.output_dir)
+
+def evaluate(cfg: DictConfig):
+    print("Starting evaluation...")
+    ppsci.utils.misc.set_random_seed(cfg.seed)
+    logger.init_logger("ppsci", osp.join(cfg.output_dir, f"{cfg.mode}.log"), "info")
+
```
同上，可删除

HydrogenSulfate commented:
@ZhangPassion pre-commit运行时需要指定--files为你的修改文件examples/perovskite_solar_cells，否则jointContribution文件夹下也会受到影响，可以通过`git checkout develop jointContribution`还原这个文件夹的改动。

![image](https://github.com/user-attachments/assets/48bb80fe-d9a6-4a66-a87f-c2d11b7e38c4)

HydrogenSulfate commented:
@ZhangPassion  完善一下PR描述，
![image](https://github.com/user-attachments/assets/beb33c32-21d0-4933-96f5-853d5743fe9c)

HydrogenSulfate commented on examples/perovskite_solar_cells/conf/psc_nn.yaml:
```diff
@@ -0,0 +1,59 @@
+
+mode: "train"
+seed: 42
+output_dir: "./output"
```
```suggestion
output_dir: ${hydra:run.dir}
```

HydrogenSulfate commented on examples/perovskite_solar_cells/conf/psc_nn.yaml:
```diff
@@ -0,0 +1,59 @@
+  lr_scheduler:
+    gamma: 0.95
+    decay_steps: 5
+  warmup_epoch: 2
+  warmup_start_lr: 1.0e-6
```
```suggestion
    warmup_epoch: 2
    warmup_start_lr: 1.0e-6
```

HydrogenSulfate commented on examples/perovskite_solar_cells/psc_nn.py:
```diff
@@ -0,0 +1,364 @@
+        epochs=train_cfg.epochs,
+        iters_per_epoch=data_size // train_cfg.batch_size,
+        learning_rate=lr,
+        gamma=0.95,
+        decay_steps=5,
+        warmup_epoch=2,
+        warmup_start_lr=1.0e-6,
```
这里请使用cfg的配置

leeleolay commented:
提交的代码对与本项目无关的代码看起来有所改动，另外没有找到相关的requirement文件，辛苦整理一下代码，也提供一下可用于模型训练的数据集

leeleolay commented:
提交的代码对其他无关代码看起来有所改动，另外没有找到相关的requirement文件

ZcwDev commented:
> 提交的代码对与本项目无关的代码看起来有所改动，另外没有找到相关的requirement文件，辛苦整理一下代码，也提供一下可用于模型训练的数据集

数据集：
通过网盘分享的文件：data.zip
链接: https://pan.baidu.com/s/17Cob37nbYVUvDbVxxznhHg?pwd=g8ip 提取码: g8ip 
--来自百度网盘超级会员v4的分享

leeleolay commented:
<img width="965" alt="image" src="https://github.com/user-attachments/assets/2f0bdf00-51f0-4524-a83d-8e4c743ac724" />
文档加载过程中有个warning，辛苦检查一下路径

leeleolay commented:
<img width="245" alt="截屏2025-04-17 11 44 45" src="https://github.com/user-attachments/assets/446e747c-0b69-4ce2-891e-be757f86fbe6" />
这里的案例名称建议修改为和文档一致的名称，比如psc_nn

leeleolay commented:
<img width="947" alt="image" src="https://github.com/user-attachments/assets/fcd5d3d9-95f5-41c4-b949-0d2d4ebeecad" />
文档这里的代码显示有点问题

leeleolay commented:
同上
<img width="989" alt="image" src="https://github.com/user-attachments/assets/9511ccb6-bcfa-456e-805f-8d271475893f" />

leeleolay commented:
文档里的requirements.txt是Conda 环境导出文件，建议修改为pip的，另外里面只需保留必要的安装依赖项即可，项目看起来用到了optuna，但是安装依赖的文档里没有囊括进来。代码运行训练时，config里只设置了10个epoch，但是程序运行时会迭代重复循环，辛苦检查一下代码逻辑。

leeleolay commented:
<img width="934" alt="image" src="https://github.com/user-attachments/assets/c07b5815-b353-4447-8d44-c318efb4b567" />
评估模型也出错了，辛苦检查一下错误

ZcwDev commented:
> <img alt="image" width="934" src="https://private-user-images.githubusercontent.com/50223303/434627149-c07b5815-b353-4447-8d44-c318efb4b567.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUzMTgwNDUsIm5iZiI6MTc0NTMxNzc0NSwicGF0aCI6Ii81MDIyMzMwMy80MzQ2MjcxNDktYzA3YjU4MTUtYjM1My00NDQ3LThkNDQtYzMxOGVmYjRiNTY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIyVDEwMjkwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhZTlkNjczYjUzNzFhNTk4Yzg0YTFkOTJiMjMyMTUyZGQ4YzljN2M3Y2FmMTI2MmEyMDY5ZWE2NmI2MmFmZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aM4P8akRChYE_NGcPQSUNYzmA9WNpTW4LNUWrgwIBKs"> 评估模型也出错了，辛苦检查一下错误
![1745317665892](https://github.com/user-attachments/assets/d01ec053-4ed7-405d-b525-694a2abb2a04)
我的运行没问题呀

leeleolay commented:
提交的代码有合入冲突，辛苦解决一下吧。
不能强制推送，需要手动解决每个冲突

ZcwDev commented:
<img width="461" alt="image" src="https://github.com/user-attachments/assets/9a5422c6-144a-4a11-b7dc-7d82bee94701" />
最近一直碰到这个问题，网上资料也试过了，SSH Key也试过了，一直提交不了，请问怎么办

ZcwDev committed:
merge code of upstream

ZcwDev committed:
Update arch.md

ZcwDev committed:
merge code of upstream

leeleolay commented:
<img width="1175" alt="image" src="https://github.com/user-attachments/assets/62ad6e95-5659-4ffc-ac7e-d17865ac90ac" />
模型评估命令运行有问题。模型训练命令，模型会反复运行，停止不了，检查一下运行逻辑。

leeleolay commented:
<img width="241" alt="image" src="https://github.com/user-attachments/assets/b8a358ff-6ab8-4289-a1f5-2205806b1682" />
requirements这里面的paddle版本的制定，我测试是cpu也可以的，是否一定限制gpu的3.0.0的版本哈，可以考虑注释掉这行或者再修改

leeleolay commented:
https://paddle-org.bj.bcebos.com/paddlescience%2Fdatasets%2Fpsc%2Fdata.zip
可以在文档中添加数据集的下载链接，方便用户使用

ZcwDev commented:
> <img alt="image" width="1175" src="https://private-user-images.githubusercontent.com/50223303/445907247-62ad6e95-5659-4ffc-ac7e-d17865ac90ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg4NjA1OTYsIm5iZiI6MTc0ODg2MDI5NiwicGF0aCI6Ii81MDIyMzMwMy80NDU5MDcyNDctNjJhZDZlOTUtNTY1OS00ZmZjLWFjN2UtZDE3ODY1YWM5MGFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjAyVDEwMzEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwMTQ0MDg4MWJlMzFiNzczNGYzYTA0Y2U3YWU5NWZiNzg2YjAxNGU1NGQ0YzczYjI4Nzc0OWRjMzQ0MjdjZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VwZEMsEFIgpOIdg6iPVvmV7DstadjLCmC7gW_NoF-dw"> 模型评估命令运行有问题。模型训练命令，模型会反复运行，停止不了，检查一下运行逻辑。

<img width="1068" alt="image" src="https://github.com/user-attachments/assets/c2d3a653-9f78-476e-867e-827d24045e4c" />
我用GPU运行的很快，你cpu可能要多等一会，就出结果了，验证的参数我换了一个，现在应该可以了（之前忘记换了）

ZcwDev commented:
要是能运行的话，帮我之前参数的下载路径也换成最新的吧谢谢

ZcwDev committed:
update

ZcwDev committed:
Merge branch 'PaddlePaddle:develop' into dev_model

leeleolay commented:
> > <img alt="image" width="1175" src="https://private-user-images.githubusercontent.com/50223303/445907247-62ad6e95-5659-4ffc-ac7e-d17865ac90ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg4NjA1OTYsIm5iZiI6MTc0ODg2MDI5NiwicGF0aCI6Ii81MDIyMzMwMy80NDU5MDcyNDctNjJhZDZlOTUtNTY1OS00ZmZjLWFjN2UtZDE3ODY1YWM5MGFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjAyVDEwMzEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwMTQ0MDg4MWJlMzFiNzczNGYzYTA0Y2U3YWU5NWZiNzg2YjAxNGU1NGQ0YzczYjI4Nzc0OWRjMzQ0MjdjZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VwZEMsEFIgpOIdg6iPVvmV7DstadjLCmC7gW_NoF-dw"> 模型评估命令运行有问题。模型训练命令，模型会反复运行，停止不了，检查一下运行逻辑。
> 
> <img alt="image" width="1068" src="https://private-user-images.githubusercontent.com/132531650/450272582-c2d3a653-9f78-476e-867e-827d24045e4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkwMzA1OTcsIm5iZiI6MTc0OTAzMDI5NywicGF0aCI6Ii8xMzI1MzE2NTAvNDUwMjcyNTgyLWMyZDNhNjUzLTlmNzgtNDc2ZS04NjdlLTgyN2QyNDA0NWU0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwNFQwOTQ0NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNTQ2OTJkMDMyMDNmNjg1NmIzOTE0NTY1ZGQ3YjZjY2ZmZjYyODY5MTcyNmZmOTRhNWFiMmIxOGIzOTI2MTc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._qAitdWQCq8Cs0noCKrCGvKNjUZ1Jd1VHv8MLB2MpbQ"> 我用GPU运行的很快，你cpu可能要多等一会，就出结果了，验证的参数我换了一个，现在应该可以了（之前忘记换了）

是训练命令，不是评估命令。训练的时候，任务不会自动终止，反复迭代的。和CPU没有关系。CPU是指在requirements中可以修改一下，原先的指定了gpu

leeleolay commented:
> 要是能运行的话，帮我之前参数的下载路径也换成最新的吧谢谢

这个辛苦你这边修改一下吧，数据的链接可以在文档的前面标记一下链接。

HydrogenSulfate commented on docs/zh/api/arch.md:
请不要修改无关文件，合入develop最新代码可解决

HydrogenSulfate reviewed (commented):
你提交的代码文件请使用pre-commit格式化

ZcwDev commented:
> > > <img alt="image" width="1175" src="https://private-user-images.githubusercontent.com/50223303/445907247-62ad6e95-5659-4ffc-ac7e-d17865ac90ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg4NjA1OTYsIm5iZiI6MTc0ODg2MDI5NiwicGF0aCI6Ii81MDIyMzMwMy80NDU5MDcyNDctNjJhZDZlOTUtNTY1OS00ZmZjLWFjN2UtZDE3ODY1YWM5MGFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjAyVDEwMzEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwMTQ0MDg4MWJlMzFiNzczNGYzYTA0Y2U3YWU5NWZiNzg2YjAxNGU1NGQ0YzczYjI4Nzc0OWRjMzQ0MjdjZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VwZEMsEFIgpOIdg6iPVvmV7DstadjLCmC7gW_NoF-dw"> 模型评估命令运行有问题。模型训练命令，模型会反复运行，停止不了，检查一下运行逻辑。
> > 
> > 
> > <img alt="image" width="1068" src="https://private-user-images.githubusercontent.com/132531650/450272582-c2d3a653-9f78-476e-867e-827d24045e4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkwMzA1OTcsIm5iZiI6MTc0OTAzMDI5NywicGF0aCI6Ii8xMzI1MzE2NTAvNDUwMjcyNTgyLWMyZDNhNjUzLTlmNzgtNDc2ZS04NjdlLTgyN2QyNDA0NWU0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwNFQwOTQ0NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNTQ2OTJkMDMyMDNmNjg1NmIzOTE0NTY1ZGQ3YjZjY2ZmZjYyODY5MTcyNmZmOTRhNWFiMmIxOGIzOTI2MTc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._qAitdWQCq8Cs0noCKrCGvKNjUZ1Jd1VHv8MLB2MpbQ"> 我用GPU运行的很快，你cpu可能要多等一会，就出结果了，验证的参数我换了一个，现在应该可以了（之前忘记换了）
> 
> 是训练命令，不是评估命令。训练的时候，任务不会自动终止，反复迭代的。和CPU没有关系。CPU是指在requirements中可以修改一下，原先的指定了gpu

图片上的就是我的训练完成结果，确实是停止了啊

Commit:
docs: fix markdown line numbers after code formatting

- Update line number references in perovskite_solar_cells_nn.md
- Align documentation with formatted psc_nn.py code structure

Commit:
Merge branch 'upstream_develop' into dev_model

leeleolay commented:
辛苦贴一下训练的log文件

leeleolay commented:
评估命令，指定路径在实际运行时报错，这个链接我测试是可以下载的，辛苦再检查一下吧
<img width="1239" alt="image" src="https://github.com/user-attachments/assets/604a268a-1701-418b-bf3c-6f0ed0be8b21" />

（或使用远程预训练模型
python psc_nn.py mode=eval eval.pretrained_model_path="https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams"）

ZcwDev commented:
> 评估命令，指定路径在实际运行时报错，这个链接我测试是可以下载的，辛苦再检查一下吧 <img alt="image" width="1239" src="https://private-user-images.githubusercontent.com/50223303/459197907-604a268a-1701-418b-bf3c-6f0ed0be8b21.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE5NjAyMDQsIm5iZiI6MTc1MTk1OTkwNCwicGF0aCI6Ii81MDIyMzMwMy80NTkxOTc5MDctNjA0YTI2OGEtMTcwMS00MThiLWJmM2MtNmYwZWQwYmU4YjIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA4VDA3MzE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwNGE2YjllZGJlYzYxM2ZmMjk0MTcxZDBkNzM0YjI5MjViMDFhNjNmYjJjZDQ0MWI4MjJiNGI4NTI1ZjFjYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.waakFp6BCaYXhUV6Ub_q2vdxv6rA-Zsj1oM7k52KHuU">
> 
> （或使用远程预训练模型 python psc_nn.py mode=eval eval.pretrained_model_path="[https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams"）](https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams%22%EF%BC%89)

评估的参数已经被我换了，那个文件就是最新文件，链接是老文件的url，上次就麻烦您帮我换成最新的文件

leeleolay commented:
> > 评估命令，指定路径在实际运行时报错，这个链接我测试是可以下载的，辛苦再检查一下吧 <img alt="image" width="1239" src="https://private-user-images.githubusercontent.com/50223303/459197907-604a268a-1701-418b-bf3c-6f0ed0be8b21.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE5NjAyMDQsIm5iZiI6MTc1MTk1OTkwNCwicGF0aCI6Ii81MDIyMzMwMy80NTkxOTc5MDctNjA0YTI2OGEtMTcwMS00MThiLWJmM2MtNmYwZWQwYmU4YjIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA4VDA3MzE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwNGE2YjllZGJlYzYxM2ZmMjk0MTcxZDBkNzM0YjI5MjViMDFhNjNmYjJjZDQ0MWI4MjJiNGI4NTI1ZjFjYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.waakFp6BCaYXhUV6Ub_q2vdxv6rA-Zsj1oM7k52KHuU">
> > （或使用远程预训练模型 python psc_nn.py mode=eval eval.pretrained_model_path="[https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams"）](https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams%22%EF%BC%89)
> 
> 评估的参数已经被我换了，那个文件就是最新文件，链接是老文件的url，上次就麻烦您帮我换成最新的文件

最新的文件是哪个

ZcwDev commented:
> > > 评估命令，指定路径在实际运行时报错，这个链接我测试是可以下载的，辛苦再检查一下吧 <img alt="image" width="1239" src="https://private-user-images.githubusercontent.com/50223303/459197907-604a268a-1701-418b-bf3c-6f0ed0be8b21.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE5NjAyMDQsIm5iZiI6MTc1MTk1OTkwNCwicGF0aCI6Ii81MDIyMzMwMy80NTkxOTc5MDctNjA0YTI2OGEtMTcwMS00MThiLWJmM2MtNmYwZWQwYmU4YjIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA4VDA3MzE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwNGE2YjllZGJlYzYxM2ZmMjk0MTcxZDBkNzM0YjI5MjViMDFhNjNmYjJjZDQ0MWI4MjJiNGI4NTI1ZjFjYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.waakFp6BCaYXhUV6Ub_q2vdxv6rA-Zsj1oM7k52KHuU">
> > > （或使用远程预训练模型 python psc_nn.py mode=eval eval.pretrained_model_path="[https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams"）](https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams%22%EF%BC%89)
> > 
> > 
> > 评估的参数已经被我换了，那个文件就是最新文件，链接是老文件的url，上次就麻烦您帮我换成最新的文件
> 
> 最新的文件是哪个

examples/perovskite_solar_cells/solar_cell_pretrained.pdparams这一个

leeleolay commented:
https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams
权重已经更新链接，可参考这个。提交的代码请不要放模型权重

leeleolay commented:
评估命令辛苦再检查一下是否能正确运行吧，我这里测试跑不起来，可以看下大小写等等之类的问题

ZcwDev commented:
> 评估命令辛苦再检查一下是否能正确运行吧，我这里测试跑不起来，可以看下大小写等等之类的问题

请问具体报错是啥呢，我这里可以跑起来

leeleolay commented:
<img width="958" height="385" alt="image" src="https://github.com/user-attachments/assets/624076ce-b3fc-4e13-afab-c150f3e9d297" />
python psc_nn.py mode=eval eval.pretrained_model_path="https://paddle-org.bj.bcebos.com/paddlescience/models/PerovskiteSolarCells/solar_cell_pretrained.pdparams"
这个命令跑不通

Commit:
Update psc_nn.py and fix documentation code snippets

- Enhanced psc_nn.py with URL model loading functionality
- Fixed all code snippet line numbers in perovskite_solar_cells_nn.md
- Updated RMSE metric in documentation to 3.91798
- Applied code formatting fixes

Commit:
Merge upstream/develop into dev_model

- Resolved merge conflict in mkdocs.yml
- Integrated new examples: Battery_LI and ML2DDB
- Kept existing psc_NN documentation

ZcwDev commented:
抱歉，才发现之前的通过 url 获取远程的参数功能有问题，现在应该没问题了，帮我再次更新一下参数：
通过网盘分享的文件：solar_cell_pretrained.pdparams
链接: https://pan.baidu.com/s/1u4jRsDUXAVlnzDgo-aGsqQ?pwd=9t4j 提取码: 9t4j 
谢谢！

leeleolay reviewed (approved):
可合入

HydrogenSulfate commented:
@ZcwDev 请使用pre-commit格式化一下本PR里的文件，否则code-style-check会失败

Commit:
merge code of upstream

ZcwDev commented:
已用pre-commit格式化，就是不知道需不需要把图片改成 URL 读取：docs/zh/examples/psc_nn_overview.png

HydrogenSulfate commented on docs/zh/api/arch.md:
```diff
@@ -40,8 +40,5 @@
         - TGCN
-        - RegDGCNN
-        - RegPointNet
-        - IFMMLP
```
这里麻烦还原回去吧，应该与你的PR无关

HydrogenSulfate commented on docs/zh/examples/psc_nn_overview.png:
已上传：<https://paddle-org.bj.bcebos.com/paddlescience/docs/psc_nn/psc_nn_overview.png>，请删除这个图片并在文档里用上面的url替换

HydrogenSulfate commented on examples/perovskite_solar_cells/conf/psc_nn.yaml:
```diff
@@ -0,0 +1,60 @@
+    warmup_epoch: 2
+    warmup_start_lr: 1.0e-6
+
+eval:
```
```suggestion
EVAL:
```

HydrogenSulfate commented on examples/perovskite_solar_cells/psc_nn.py:
```diff
@@ -0,0 +1,412 @@
+
+        train_constraint = create_constraint(train_dict, cfg.TRAIN.batch_size)
+        verif_validator = create_validator(
+            verif_dict, cfg.eval.batch_size, "verif_validator"
```
`cfg.eval.`请替换成`cfg.EVAL.`

HydrogenSulfate reviewed (commented):
有几点小问题，辛苦修改下，其它应该没有了

Commit:
merge code of upstream

Commit:
docs: update arch.md with new models
